### PR TITLE
Added cylinder primitive

### LINKED ISSUE #318: Cylinder primitive generation
It would be good to have a function that generates the vertices and triangles of a cylinder. 

Primitives examples are available [here](https://github.com/fury-gl/fury/blob/master/fury/primitive.py).

### PR DESCRIPTION
**Pull request by tushar5526**
Closes #318 
Referred to this [source](http://songho.ca/opengl/gl_cylinder.html) for generating vertices and triangles.
Formatted the `primitive.py` according to PEP 8.

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Thank you @tushar5526 for this primitive!

we will look into it and give you more detailed comments. But for now, can you add some unit tests? You can look at https://github.com/fury-gl/fury/blob/master/fury/tests/test_primitive.py. 

Thank you.

**Comment by tushar5526** [comment]
Thank you for reviewing. Will add the tests soon :)

**Comment by tushar5526** [comment]
@skoudoro, I am a little confused about how to implement the tests for this, asserting the generated values to hardcoded numbers does not seems to be an option here.
PS: I have never written tests before so if there is anything I need to keep in mind, do tell me.
Thanks

**Comment by skoudoro** [comment]
Overall, it looks good. It will be good if you could change the default orientation of the cylinder (length along Y instead of Z).
See below:

Capture d’écran 2020-10-07 à 02 42 23

I created a PR in your fork to add some tests https://github.com/tushar5526/fury/pull/2.  it is just missing a test when it is capped. I let you add this one last one.

File: fury/primitive.py [hunk 1]
```diff
@@ -670,25 +670,150 @@ def prim_frustum():
...
+                          [6, 1, 2],
+                          [1, 6, 5],
+                          [5, 0, 1],
+                          [0, 5, 4]], dtype='u8')
     vertices /= 2
     triangles = fix_winding_order(vertices, triangles, clockwise=True)
     return vertices, triangles
+
+
+def prim_cylinder(radius=1, height=3, sectors=36, capped=True):
```
Comment (skoudoro) [inline fury/primitive.py:697]: Could you set by default:  `radius=0.5, height=1`

File: fury/primitive.py [hunk 2]
```diff
@@ -670,25 +670,150 @@ def prim_frustum():
...
+    """Return vertices and triangle for a cylinder.
```
Comment (skoudoro) [inline fury/primitive.py:698]: typo: `triangles`

File: fury/primitive.py [hunk 3]
```diff
@@ -670,25 +670,150 @@ def prim_frustum():
...
+
+    Returns
+    -------
+    vertices: ndarray
+        vertices coords that compose our cylinder
+    triangles: ndarray
+        triangles that compose our cylinder
+    """
+
+    if not isinstance(sectors, int):
```
Comment (skoudoro) [inline fury/primitive.py:719]: check also if sectors are meaningful:  

`if not isinstance(sectors, int) and sectors < 8:`

**Comment by tushar5526** [comment]
Cylinder's base is on XZ plane now and the axes on Y.
Reproduced the mesh in unity for testing.
Screenshot from 2020-10-07 17-13-25

**Review by skoudoro — APPROVED** [review]
This PR looks good to me and is ready to go. 

I will wait for any additional comments until tomorrow and then merge it.

**Comment by skoudoro** [comment]
Thank you @tushar5526

**Pull request merged**